sidechain/storage/src/interface.rs

```diff
@@ -52,7 +74,17 @@ impl<SignedBlock: SignedBlockT> BlockStorage<SignedBlock> for SidechainStorageLo
 }
 
 impl<SignedBlock: SignedBlockT> BlockPruner for SidechainStorageLock<SignedBlock> {
-	fn prune_blocks_except(&self, blocks_to_keep: BlockNumber) {
-		self.storage.write().prune_shards(blocks_to_keep);
+	fn prune_blocks_except(&self, number_of_blocks_to_keep: BlockNumber) {
+		self.storage.write().prune_shards(number_of_blocks_to_keep);
+	}
+}
+
+impl<SignedBlock: SignedBlockT> FetchBlocks<SignedBlock> for SidechainStorageLock<SignedBlock> {
+	fn fetch_all_blocks_following(
+		&self,
+		block_hash: &BlockHash,
+		shard_identifier: &ShardIdentifierFor<SignedBlock>,
+	) -> Result<Vec<SignedBlock>> {
+		self.storage.read().get_blocks_following(block_hash, shard_identifier)
```

Comment:
Implementation is also just a wrapper, like the other `impls` of the `SidechainStorageLock`.